chore: remove unreferenced devDeps

nschonni opened the pull request:
Few devDeps that don't appear used right now

K-9196 commented:
![IMG_20210414_231434_419](https://user-images.githubusercontent.com/132248353/235412381-d890255d-93d8-4586-ab21-04eb3c8eea6d.jpg)
Snapchat

isaacs commented on package.json:
```diff
@@ -57,11 +57,9 @@
     "endOfLine": "lf"
   },
   "devDependencies": {
-    "@types/brace-expansion": "^1.1.0",
```
This is necessary for typescript compilation.

nschonni replied:
I didn't get any warnings/errors from `tsc` as this package is only a subdep of tap, but I added it back anyway

isaacs replied:
Oh! lol, I thought this was minimatch, sorry I guess I have too many modules 😅 Yes, that can be removed. I'll get this merged, don't worry about updating a second time.

isaacs reviewed (changes requested):
@types/brace-expansion needs to stay

nschonni committed:
chore: remove unreferenced devDeps

PR-URL: https://github.com/isaacs/node-mkdirp/pull/44
Credit: @nschonni
Close: #44
Reviewed-by: @isaacs
